packages/ra-ui-materialui/src/auth/Login.tsx

```diff
@@ -8,49 +13,48 @@ import {
     createMuiTheme,
     withStyles,
     createStyles,
+    WithStyles,
+    Theme,
 } from '@material-ui/core/styles';
 import LockIcon from '@material-ui/icons/Lock';
+import { StaticContext } from 'react-router';
 
 import defaultTheme from '../defaultTheme';
 import Notification from '../layout/Notification';
 import DefaultLoginForm from './LoginForm';
 
-const styles = theme => createStyles({
-    main: {
-        display: 'flex',
-        flexDirection: 'column',
-        minHeight: '100vh',
-        height: '1px',
-        alignItems: 'center',
-        justifyContent: 'flex-start',
-        backgroundRepeat: 'no-repeat',
-        backgroundSize: 'cover',
-    },
-    card: {
-        minWidth: 300,
-        marginTop: '6em',
-    },
-    avatar: {
-        margin: '1em',
-        display: 'flex',
-        justifyContent: 'center',
-    },
-    icon: {
-        backgroundColor: theme.palette.secondary[500],
-    },
-});
+interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
```

Comment:
we should not extend from `HtmlHTMLAttributes`, but use an union type in the component definition:

```
class LoginView extends Component<Props & WithStyles<typeof styles> & HtmlHTMLAttributes<HTMLDivElement>>
```

I think having the Props type extending nothing makes the intent of the component clearer.

Reply:
Why not